technology/cache/src/main/java/com/github/mizool/technology/jcache/timeouting/TimeoutingExecutor.java

```diff
@@ -41,51 +33,16 @@
         .read()
         .orElse(10000L);
 
-    private final ExecutorService executorService;
+    private final ExecutorService executorService = Executors.newWorkStealingPool();
 
     public <T> T execute(Callable<T> callable)
     {
-        try
-        {
-            Future<T> future = executorService.submit(callable);
-            return future.get(TimeoutingExecutor.CACHE_TIMEOUT, TimeUnit.MILLISECONDS);
-        }
-        catch (TimeoutException e)
-        {
-            throw new CacheTimeoutException(e);
-        }
-        catch (InterruptedException e)
-        {
-            Thread.currentThread()
-                .interrupt();
-            throw new UncheckedInterruptedException(e);
-        }
-        catch (ExecutionException e)
-        {
-            throw new UncheckedExecutionException(e);
-        }
+        return Futures.get(executorService.submit(callable),
+            Duration.of(TimeoutingExecutor.CACHE_TIMEOUT, ChronoUnit.MILLIS));
```

Comment:
Let's catch `UncheckedTimeoutException` and wrap it in a `CacheTimeoutException`. This way, we keep the existing behavior that `SafeCacheLogHelper` relies upon.

Alternatively, we should remove `CacheTimeoutException` and change `SafeCacheLogHelper` to catch `UncheckedTimeoutException` instead.

I tend towards the second solution.